packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -966,18 +1016,32 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     String textBefore = selection.textBefore(text);
     String textAfter = selection.textAfter(text);
     int cursorPosition = math.min(selection.start, selection.end);
+    
     // If not deleting a selection, delete the next/previous character.
     if (selection.isCollapsed) {
       if (!forward && textBefore.isNotEmpty) {
-        final int characterBoundary = previousCharacter(textBefore.length, textBefore);
-        textBefore = textBefore.substring(0, characterBoundary);
-        cursorPosition = characterBoundary;
+        final bool endsWithBreakLine = textBefore.codeUnitAt(textBefore.length - 1) == 0x0A;
+        final bool hasModifier = metaModifier || altModifier;
+        
+        if (!hasModifier || endsWithBreakLine) {
```

Comment:
when the cursor is next to a break line, then any type of delete will only delete the line break.